homestar-runtime/src/cli/init.rs

```diff
@@ -0,0 +1,317 @@
+use std::{
+    fmt::Display,
+    fs::File,
+    io::{empty, stdout, Write},
+    path::PathBuf,
+    str::FromStr,
+};
+
+use clap::ValueEnum;
+use inquire::{ui::RenderConfig, Confirm, CustomType, Select};
+use miette::{bail, miette, Result};
+use rand::Rng;
+use serde::de::IntoDeserializer;
+use serde_with::{base64::Standard, formats::Padded, DeserializeAs, SerializeAs};
+
+use crate::{
+    settings::{KeyType, PubkeyConfig},
+    ExistingKeyPath, NetworkBuilder, NodeBuilder, RNGSeed, SettingsBuilder,
+};
+
+/// Where to write the resulting configuration.
+#[derive(Debug)]
+pub enum OutputMode {
+    /// Write to standard output.
+    StdOut,
+    /// Write to a file.
+    File {
+        /// The path to write to.
+        path: PathBuf,
+        /// Automatically overwrite the file if it exists.
+        force: bool,
+    },
+}
+
+#[derive(Debug)]
+enum PubkeyConfigOption {
+    GenerateFromSeed,
+    FromFile,
+}
+
+/// The arguments for configuring the key
+#[derive(Debug)]
+pub enum KeyArg {
+    /// Load the key from an existing file
+    File {
+        /// The path of the file
+        path: PathBuf,
+    },
+    /// Generate the key from a seed
+    Seed {
+        /// The base64 encoded 32 byte seed to use for key generation
+        seed: Option<String>,
+    },
+}
+
+/// The type of key to generate
+#[derive(Debug, Clone, PartialEq, ValueEnum)]
+pub enum KeyTypeArg {
```

Comment:
Throwing this out there: can we not reuse/extend KeyType from https://github.com/ipvm-wg/homestar/pull/582/files#diff-ee5055c2e074408bf2f7e2fbd583688042eaeec89be0fdeb760e7ca3cf2ef478R32?

Reply:
Not doing a deep-dive yet on pubkey_config again, but are there are more shared elements with a little bit of cleanup? Throwing that out there just as a check.

Reply:
I was torn on reusing `KeyType`, because that would give up control over which options are displayed in the initialization wizard, when I can imagine situations where we may want to support legacy key types (like RSA), but without necessarily exposing them through the first-time setup.

Happy to reuse it if you don't think that's a concern though!

Reply:
I figure that since pubkey config will always be tied to what a user can provide (i.e. settings), it'd be in sync with whichever key types we'll need to init with?